Sprint 2

DIvanov1ch opened the pull request:
1. Task: [Sprint 2](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint%232.md)
2. Screenshot:
![Sprint2](https://github.com/DIvanov1ch/eCommerce-Application/assets/93213426/7a05700f-5407-487a-ace9-abcc20c9d2cf)

3. Deploy: [netlify](https://sd-merch.netlify.app/)
4. Done 21.08.2023 / deadline 22.08.2023
5. Score: 60 / 60
 ### 👨‍🏫 Mentor Evaluation Criteria (60 points)
 ------
  - [x] - Code Review Participation (10 points): Students actively participate in code reviews, providing constructive feedback and suggestions to their peers.
  - [x] -No Merges Without Approvals (10 points): Pull requests are not merged without approvals from other team members, ensuring the quality of code contributions.
  - [x] - Pull Request Descriptions (10 points): Pull requests include detailed descriptions, outlining the changes made and the purpose of the changes, facilitating better understanding and review.
  - [x] - Task Board Management (10 points): Students maintain and regularly update a task board (e.g., Trello, Jira), ensuring that tasks are moved according to their progress and completion.
  - [x] - Active Participation in Development Discussions (5 points): Students take an active role in discussing the development process, sharing ideas, and collaborating on solutions and improvements.
  - [x] - Unit Test Coverage (15 points): The codebase has a minimum of 25% unit test coverage, ensuring the reliability and robustness of the implemented features.
 
### Penalties for Mentor Evaluation Criteria: ❌
--------
 - [x] - Linter Errors (5 points): If there are errors from the linter, a penalty of 5 points will be applied. Ensure that you follow the linter rules to maintain high code quality. 
 - [x] - Prettier Errors (5 points): If there are formatting errors detected by Prettier, a penalty of 5 points will be applied. It's important to adhere to established formatting conventions for clean, readable code
 - [x] - Presence of 'any' (5 points): The use of 'any' in TypeScript can lead to potential typing issues and is generally discouraged. A penalty of 5 points will be applied if 'any' is found in the code.
 - [x] - Non-functional Scripts in package.json (5 points): All scripts defined in the package.json should function correctly. If there are non-working scripts, a penalty of 5 points will be applied.
 - [x] - Non-SPA Application (60 points): The application should be a Single Page Application (SPA) with an empty index.html. If this requirement is not met, a penalty of 60 points will be applied.(8/8)

mercartem reviewed (approved):
Отличная организация в команде. Все пункты первого спринта выполнены. Так держать! Единственный вопрос по структуре папок. В папке components на ряду с footer и header папка ui. Зачем? Логичнее занести папку для интерфейса в папку компонента, если будете разделять логику и представление непосредственно в компонентах.

Commit:
feat: add Prices with and Without Discount for Discounted Products

Commit:
feat: make limit of cards on one page to 500

beskibeski committed:
add catalog and display product list (#100)

Commit:
feat: add basic filter bars and refactor: CatalogPage

DIvanov1ch committed:
refactor: update date id

DIvanov1ch committed:
feat: add user classes

DIvanov1ch committed:
feat: implement display of user's personal info

DIvanov1ch committed:
Merge branch 'release/catalog-product-profile' into user-profile

DIvanov1ch committed:
Present user info (#101)

DIvanov1ch committed:
feat: implement routing to user profile page

DIvanov1ch committed:
fix: add css rule limit

DIvanov1ch committed:
fix: add missing date field

phoinixru committed:
chore: update husky

phoinixru committed:
chore: prettify files

phoinixru committed:
feat: remove test login page

phoinixru committed:
feat: remove project page

phoinixru committed:
feat: register routes

phoinixru committed:
feat: pass params to router page

Commit:
feat: add filter functionality and waiting emulation

DIvanov1ch committed:
refactor: update component class name

DIvanov1ch committed:
feat: add popup templates

DIvanov1ch committed:
feat: implement edit mode

phoinixru committed:
feat: implement product page

DIvanov1ch committed:
User profile edit mode (#103)

phoinixru committed:
Merge branch 'release/catalog-product-profile' into product/routing

phoinixru committed:
Product page (#104)

phoinixru committed:
feat: implement bread-crumbs component

phoinixru committed:
feat: disable current category in crumbs

phoinixru committed:
chore: add default PR template

phoinixru committed:
Bread crumbs (#105)

DIvanov1ch committed:
refactor: extract some methods for readability

Commit:
feat: add sorting by name and price

Commit:
feat: add search input, buttons and query for search api

DIvanov1ch committed:
refactor: update html templates

DIvanov1ch committed:
refactor: update css classes and styles

DIvanov1ch committed:
refactor: remove redundant methods

DIvanov1ch committed:
feat: add delete-template

phoinixru committed:
feat: allow change category/slug for breadcrumbs

phoinixru committed:
feat: implement product categories component

phoinixru committed:
chore: fix husky

phoinixru committed:
feat: add code examples page

phoinixru committed:
Categories Component (#106)

phoinixru committed:
feat: add profile link to header

phoinixru committed:
fix: class name

phoinixru committed:
feat: add catalog link to header

phoinixru committed:
fix: handle click only for mobile

phoinixru committed:
fix: prevent redirects from toggle

phoinixru committed:
fix: styles

phoinixru committed:
fix: use unique class name

DIvanov1ch committed:
feat: add toastify-js

DIvanov1ch committed:
feat: add refresh token client and update method

DIvanov1ch committed:
fix: update is-valid-value function

DIvanov1ch committed:
feat: implement personal info and email edit

DIvanov1ch committed:
chore: update lintstaged rules

phoinixru committed:
feat: implement price-box component

phoinixru committed:
feat: show prices on product page

DIvanov1ch committed:
Merge branch 'release/catalog-product-profile' into user-profile-update-personal-info

phoinixru committed:
Header profile link (#107)

phoinixru committed:
Catalog header link (#108)

phoinixru committed:
docs: add examples for price-box

DIvanov1ch committed:
User profile personal info and email edit (#109)

phoinixru committed:
Product prices (#110)

Commit:
feat: add search highligting and multiple text queries searching

Commit:
fix: resolve merge conflict

DIvanov1ch committed:
chore: install toastify-js types

beskibeski committed:
Catalog/filter sort search (#111)

phoinixru committed:
fix: overlay position

phoinixru committed:
feat: add swiper package

phoinixru committed:
fix: update properties type

phoinixru committed:
feat: implement image slider component

phoinixru committed:
docs: add slider example to components page

phoinixru committed:
feat: use image slider on product page

phoinixru committed:
chore: remove routes folder

phoinixru committed:
feat: implement modal component

phoinixru committed:
feat: show slider in modal

phoinixru committed:
feat: use slider with modal on product page

phoinixru committed:
docs: add modal example to components page

phoinixru committed:
fix: examples page

phoinixru committed:
fix: hide params block

phoinixru committed:
Product images (#112)

DIvanov1ch committed:
feat: add change-password method to api file

DIvanov1ch committed:
refactor: remove redundant css class

Commit:
feat: add interactive product cards

Commit:
Merge branch 'release/catalog-product-profile' of https://github.com/DIvanov1ch/eCommerce-Application into catalog/interactive-product-cards

DIvanov1ch committed:
feat: add css rules for delete address button

DIvanov1ch committed:
fix: implement correct displaying input errors

DIvanov1ch committed:
feat: implement password change in user profile

DIvanov1ch committed:
Password change in user profile (#113)

Commit:
refactor: use pricebox component in catalogPage and redesign

Commit:
Merge branch 'release/catalog-product-profile' of https://github.com/DIvanov1ch/eCommerce-Application into catalog/interactive-product-cards

DIvanov1ch committed:
refactor: replace template strings with function

DIvanov1ch committed:
fix: implement correct password validation

Commit:
feat: add breadcrumbs component and delete slogan

DIvanov1ch committed:
feat: implement adding and deleting addresses

beskibeski committed:
Catalog/interactive product cards (#114)

phoinixru committed:
feat: filter products by category

DIvanov1ch committed:
feat: implement manage addresses

phoinixru committed:
Catalog category filter (#115)

DIvanov1ch committed:
Manage addresses in user profile (#116)

DIvanov1ch committed:
Release/catalog product profile (#117)

DIvanov1ch committed:
fix: add warning icons

DIvanov1ch committed:
fix: correct current password validation

phoinixru committed:
fix: storing data in local storage

phoinixru committed:
refactor: move function to utils

phoinixru committed:
fix: loading categories

phoinixru committed:
feat: implement product card component

phoinixru committed:
feat: use product card on catalog page

phoinixru committed:
Productcard (#118)

phoinixru committed:
Product card (#139)

phoinixru committed:
fix: sorting / filtering

phoinixru committed:
fix: sorting / filtering (#140)

DIvanov1ch committed:
fix: refresh token flow

phoinixru committed:
feat: make app PWA

phoinixru committed:
feat: make header sticky

phoinixru committed:
feat: make slider button more pronounced

phoinixru committed:
feat: use material icons

phoinixru committed:
refactor: catalog page

Commit:
feat: add cart button on product cards

Commit:
refactor: make bigger inactive cart items

beskibeski committed:
interactive product cards with basket (#143)

phoinixru committed:
PWA (#141)

phoinixru committed:
Material icons (#142)

DIvanov1ch committed:
feat: add loadCustomer() before render profile

DIvanov1ch committed:
refactor: add update-actions enum

DIvanov1ch committed:
refactor: optimize function length

phoinixru committed:
feat: add styles for mobile page

DIvanov1ch committed:
fix: remove redundant user field from Store

DIvanov1ch committed:
refactor: update LoggedInUser class

DIvanov1ch committed:
refactor: move idSelector to utils

DIvanov1ch committed:
refactor: replace querySelector with $

DIvanov1ch committed:
refactor: optimize registration page file

phoinixru committed:
feat: show variants on detail page

phoinixru committed:
feat: add styles for mobile page (#144)

DIvanov1ch committed:
fix: add warning icon to html file

DIvanov1ch committed:
refactor: update css classes

DIvanov1ch committed:
refactor: update registration page

DIvanov1ch committed:
refactor: update profile page

DIvanov1ch committed:
Merge branch 'release/basket-about_us' into user-profile-fix-refresh-token

DIvanov1ch committed:
feat: add country constant

DIvanov1ch committed:
fix: replace className with classSelector

DIvanov1ch committed:
fix: rename icon class to image class

DIvanov1ch committed:
Fix bugs on profile and registration pages (#145)

DIvanov1ch committed:
refactor: move service functions to utils

DIvanov1ch committed:
feat: add Validator class

DIvanov1ch committed:
Reg profile clean code (#146)

Commit:
feat: add anonymous flow and cart functions

Commit:
fix: merge conflict

Commit:
feat: add some functionality for creating carts and adding products

Commit:
feat: add full functionality for adding products in anonymous and login users

DIvanov1ch committed:
refactor: move update-actions to enums

DIvanov1ch committed:
refactor: rename get-toast-options()

DIvanov1ch committed:
refactor: update error-messages

DIvanov1ch committed:
feat: add popup component

DIvanov1ch committed:
feat: add addAddress component

DIvanov1ch committed:
feat: add changePassword component

DIvanov1ch committed:
feat: add deleteAddress component

DIvanov1ch committed:
feat: add editAddress component

DIvanov1ch committed:
feat: add editProfile component

DIvanov1ch committed:
refactor: user profile page major fix

DIvanov1ch committed:
User profile refactoring (#147)

DIvanov1ch committed:
refactor: update loadCustomer function

DIvanov1ch committed:
fix: eventListener bug

Commit:
feat: add merging anonymous cart with user cart

Commit:
feat: add loader indicator

Commit:
fix: registration login bug

beskibeski committed:
Integrate with api cart (#148)

DIvanov1ch committed:
feat: add CartCard component

DIvanov1ch committed:
feat: add CustomerCart and Price classes

DIvanov1ch committed:
feat: add getCart() to API file

DIvanov1ch committed:
feat: add CartPage

phoinixru committed:
chore: merge 'release/basket-about_us' into product/variants

DIvanov1ch committed:
feat: add customerCart in Store object

phoinixru committed:
feat: add product variant to cart

DIvanov1ch committed:
feat: implement Cart page

DIvanov1ch committed:
Merge branch 'release/basket-about_us' into basket/display-basket-items

phoinixru committed:
Product variants (#149)

DIvanov1ch committed:
refactor: update function names

DIvanov1ch committed:
Merge branch 'release/basket-about_us' into basket/display-basket-items

DIvanov1ch committed:
Basket/display basket items (#150)

DIvanov1ch committed:
fix: update user check before going to profile

DIvanov1ch committed:
fix: update put-product-to-cart()

DIvanov1ch committed:
fix: add missing argument

DIvanov1ch committed:
feat: add variantId

phoinixru committed:
fix: menu toggle

DIvanov1ch committed:
refactor: change const name

phoinixru committed:
fix: navigation items alignment

phoinixru committed:
Merge remote-tracking branch 'origin/put-product-into-cart' into catalog/pagination

phoinixru committed:
feat: implement infinite scroll on catalog page

phoinixru committed:
fix: catalog page style

phoinixru committed:
feat: add icons to links in header

DIvanov1ch committed:
Put product into cart (#151)

DIvanov1ch committed:
feat: add item-counter component

DIvanov1ch committed:
feat: implement modify product quantity

phoinixru committed:
Catalog product loading (#152)

DIvanov1ch committed:
Basket/modify product quantity (#153)

DIvanov1ch committed:
feat: implement remove product from cart

DIvanov1ch committed:
feat: implement remove product from cart (#154)

phoinixru committed:
feat: implement team member component

phoinixru committed:
feat: add about us page

phoinixru committed:
Merge branch 'release/basket-about_us' into about_us

phoinixru committed:
feat: add rsschool logo to about us page

Commit:
fix: anonymousId bug and active product carts after merging

phoinixru committed:
feat: show badge with number of items in cart

Commit:
feat: add Product to Cart Functionality for Non-Duplicated Products

Commit:
feat: add Remove Product from Cart Functionality and loader text

Commit:
refactor: change cart by customerCart

beskibeski committed:
Update index.ts

add my bio

DIvanov1ch committed:
feat: add info about myself

phoinixru committed:
feat: show badge with number of items in cart (#156)

beskibeski committed:
Product/add remove product (#157)

DIvanov1ch committed:
refactor: put CartUpdateActions to enum

DIvanov1ch committed:
refactor: change custom event names

DIvanov1ch committed:
feat: implement message for empty cart

DIvanov1ch committed:
feat: add clear-shopping-cart button

DIvanov1ch committed:
fix: implement correct redirection after login

DIvanov1ch committed:
feat: add empty-cart image

DIvanov1ch committed:
feat: implement clear-cart button

phoinixru committed:
feat: add icon util function

phoinixru committed:
feat: add cart actions utils

phoinixru committed:
feat: handle add/remove item on product page

DIvanov1ch committed:
Basket/display empty cart message link to catalog (#158)

phoinixru committed:
feat: color buttons

beskibeski committed:
Update template.html

phoinixru committed:
Product/add to cart button (#159)

DIvanov1ch committed:
refactor: delete unused function

DIvanov1ch committed:
refactor: delete cart-merge file

phoinixru committed:
Merge branch 'release/basket-about_us' into about_us

phoinixru committed:
About us (#155)

DIvanov1ch committed:
refactor: delete unused function (#160)

phoinixru committed:
fix: text on about us page

DIvanov1ch committed:
refactor: update cart methods

DIvanov1ch committed:
fix: close dialog after removing items

DIvanov1ch committed:
feat: implement promo code feature

DIvanov1ch committed:
feat: implement promo ticket

DIvanov1ch committed:
feat: implement updated prices

DIvanov1ch committed:
feat: add SAVE15 promo code

DIvanov1ch committed:
fix: correct removing promo code

DIvanov1ch committed:
Basket/promo code (#161)

phoinixru committed:
feat: add promo codes to home page

phoinixru committed:
fix: typo

phoinixru committed:
fix: disable controls while changing count

phoinixru committed:
feat: display variant name

phoinixru committed:
fix: disable controls on input change

phoinixru committed:
Promo/home copy (#162)

phoinixru committed:
fix: update photo

DIvanov1ch committed:
Sprint 4 (#163)

beskibeski committed:
feat: update photo

beskibeski committed:
feat: update photo (#164)

dependabot[bot] committed:
chore(deps-dev): bump @babel/traverse from 7.22.8 to 7.23.2

Bumps [@babel/traverse](https://github.com/babel/babel/tree/HEAD/packages/babel-traverse) from 7.22.8 to 7.23.2.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.23.2/packages/babel-traverse)

---
updated-dependencies:
- dependency-name: "@babel/traverse"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

DIvanov1ch committed:
chore(deps-dev): bump @babel/traverse from 7.22.8 to 7.23.2 (#166)
